shared/p2p/service_test.go

```diff
@@ -70,15 +71,101 @@ func TestP2pPortTakenError(t *testing.T) {
 }
 
 func TestBroadcast(t *testing.T) {
-	s, err := NewServer(&ServerConfig{})
+	servers, err := runP2PServersWithDifferentPorts(3)
 	if err != nil {
-		t.Fatalf("Could not start a new server: %v", err)
+		t.Fatalf("error while trying to create server: %s", err)
+	}
+
+	err = connectServersTo(servers[0], servers[1:])
+	if err != nil {
+		t.Fatalf("error while trying to connect to peer: %s", err)
 	}
 
 	msg := &shardpb.CollationBodyRequest{}
-	s.Broadcast(msg)
+	subscribeServersToTopic(servers, "theTopic", msg)
+
+	time.Sleep(1 * time.Second)
```

Comment:
Why sleep? Can we remove this?

Reply:
I tried it, but since calling RegisterTopic executes a goroutine I assume that it takes it's time until it is registered (maybe distributed in all nodes?). Removing the sleep makes it not broadcast and not pass. Any advice in this?

Reply:
Sorry @prestonvanloon can we look at this with a little bit of love? :D I still did not get a good response, take a look at this line in the current code:

https://github.com/prysmaticlabs/prysm/blob/074ac07dc8b2cd95cfa1a6764ab0852e4aac10f2/shared/p2p/service_test.go#L159

Looks like this problem for sleep was detected long ago but not resolved. I think in order to have more control on when a topic is registered the service needs to be refactored a little and it needs to be done in another issue!

Do you have another suggestion?

Reply:
What is the minimum time we can set for this to work? Can it be short like the 10ms example you provided? I think I can live with that.

Reply:
+1, try experimenting with as low of a value as you can

Reply:
@prestonvanloon @rauljordan finally with 20 milliseconds is enough to connect 3 nodes between them for the test.